File: tests/test_capability_router.py
"""Tests for capability graph router."""

from coordination_patterns.capability_router.pattern import AgentRouter, RoutingIntent


def test_route_known_action():
    router = AgentRouter()
    intent = RoutingIntent(
        action="find",
        resource="sales_report",
        parameters={"date_range": "Q1-2026"},
    )
    result = router.route_request(intent)
    assert result == "SalesAgent"


def test_route_unknown_action():
    router = AgentRouter()
    intent = RoutingIntent(
        action="analyze",
        resource="server_log",
        parameters={},
    )
    result = router.route_request(intent)
    assert "Error" in result


def test_routing_intent_validation():
    """RoutingIntent rejects invalid actions/resources."""
    import pytest
    from pydantic import ValidationError

    with pytest.raises(ValidationError):
        RoutingIntent(
            action="delete",  # not a valid ActionType
            resource="sales_report",
            parameters={},
        )

File: tests-unit/test_capability_router.rs
use local_agent::capability_router::{ActionType, AgentRouter, ResourceType, RoutingIntent};
use serde_json::json;

#[test]
fn test_route_known_action() {
    let router = AgentRouter::new();
    let intent = RoutingIntent {
        action: ActionType::Find,
        resource: ResourceType::SalesReport,
        parameters: json!({"date_range": "Q1-2026"}),
    };
    let result = router.route_request(&intent);
    assert_eq!(result, "SalesAgent");
}

#[test]
fn test_route_unknown_action() {
    let router = AgentRouter::new();
    let intent = RoutingIntent {
        action: ActionType::Analyze,
        resource: ResourceType::ServerLog,
        parameters: json!({}),
    };
    let result = router.route_request(&intent);
    assert!(result.contains("Error"));
}

#[test]
fn test_routing_intent_serialization() {
    let intent = RoutingIntent {
        action: ActionType::Find,
        resource: ResourceType::SalesReport,
        parameters: json!({"quarter": "Q1"}),
    };
    let serialized = serde_json::to_string(&intent).unwrap();
    let deserialized: RoutingIntent = serde_json::from_str(&serialized).unwrap();
    assert_eq!(intent.action, deserialized.action);
    assert_eq!(intent.resource, deserialized.resource);
    assert_eq!(intent.parameters, deserialized.parameters);
}
